crates/services/upgradable-executor/src/executor.rs

```diff
@@ -225,16 +307,48 @@ where
     where
         TxSource: TransactionsSource + Send + Sync + 'static,
     {
-        #[cfg(feature = "wasm-executor")]
-        return self.wasm_execute_inner(block, options);
+        let block_version = block.state_transition_version();
+        if block_version == Self::VERSION {
+            match &self.execution_strategy {
+                ExecutionStrategy::Native => self.native_execute_inner(block, options),
+                ExecutionStrategy::Wasm { module } => {
+                    self.wasm_execute_inner(module, block, options)
+                }
+            }
+        } else {
+            let module = self.get_module(block_version)?;
```

Comment:
Do we have a way to ensure that the initial version of the executor is available in wasm? For example, if there have been several upgrades to the network and a node on the latest native version attempts to sync the network, will the bytecode for the initial executor version (before any upgrades occured) be available?

Reply:
No, we don't have a way. It should already be part of the snapshot/genesis block.

Reply:
Automated the process of insertion of the Wasm bytecode into the `ChainConfig` https://github.com/FuelLabs/fuel-core/pull/1826